GTBitsOfGood/angels-among-us
# Onboarding Scaffolding Part 2

## Onboarding Scaffolding Part 2

Issue Number(s): #34 

Finishes the onboarding sequence by adding an intro/completion slide, tooltip on slides that need it, multi-dropdown on slides that need it, reformatting the options, allowing single select/multi-select for different questions, and printing out all selected options on submit.

Note: The ticket says to "Create a component for an introductory and completion slide." but I instead just had one component for all 3, as the layout for each is all very similar. I used the "description" field in the JS object to put the text paragraphs for the intro/completion slide and left them blank for question slides (but I think it could be useful in the future in case elaboration is needed on the questions). I also left the "options" field blank in the intro/completion slides so no options would be rendered, and this way I don't have to make extra components. I don't know if this is the best way to do it though.

### Checklist


### Critical Changes

- None

### Related PRs

- None

### Testing

Test dynamic creation of question title, description, and option:
- Change text of title and description in JS object and check that it's changed
- Add/remove/change options and check that they are changed

Test answers state:
- Check that answers can be selected and de-selected
- Check answers on question(s), move to next question(s), return to previous question(s) and check that those answers are still checked

Test single select questions:
- Check that if nothing is selected, none of the options are grayed out
- Check that if one option is selected, the other is grayed out
- Check that if one option is selected and you click on the other option, the option switches and the other options get grayed out

Test dropdown:
- Check that multiple items can be selected in the dropdown
- Check adding/removing selections in dropdown
- Check searching in dropdown

Test printing of final answers:
- Select various options on questions (including all options on a question or no options on a question) and check that the object printed out is correct
